src/lib/pipeline.ml

```diff
@@ -324,9 +328,11 @@ module Construct = struct
        make_target }
       bam_pair
 
-end
+  let rna_hla_typer fastq_sample = Rna_hla_typer fastq_sample
```

Comment:
I disagree, I think that the functions in `Pipeline.Construct` should be convenience wrappers for the elements of `Pipeline.t` GADT that they construct.

This distinction is similar, but different, from the aligner vs caller difference in the pipeline. At the moment, when you look at the variant caller "constructors"  in this module (ex `mutect`,  `muse`, `virmid`, `strelka` ... etc) they all have a similar type ( `: ~configuration:_ -> bam_pair t -> vcf t`) and a similar structure:

```OCaml
  let mutect ?(configuration=Mutect.Configuration.default) bam_pair =
    let configuration_name = configuration.Mutect.Configuration.name in
    let configuration_json = Mutect.Configuration.to_json configuration in
    let make_target
        ~reference_build ~run_with ~input ~result_prefix ~processors
        ?more_edges () =
      match input with | Variant_caller.Somatic {normal; tumor} ->
      Mutect.run
        ~configuration
        ?more_edges
        ~reference_build ~run_with
        ~normal ~tumor
        ~result_prefix `Map_reduce in
    somatic_variant_caller
      {Variant_caller.name = "Mutect";
       configuration_json;
       configuration_name;
       make_target;}
      bam_pair
``` 
vs
```OCaml
let mutect2 ?(configuration=Gatk.Configuration.Mutect2.default) bam_pair =
    let configuration_name = configuration.Gatk.Configuration.Mutect2.name in
    let configuration_json = Gatk.Configuration.Mutect2.to_json configuration in
    let make_target
        ~reference_build ~run_with ~input ~result_prefix ~processors
        ?more_edges () =
      match input with
      | Variant_caller.Somatic {normal; tumor} -> Gatk.mutect2
        ~configuration
        ?more_edges
        ~reference_build ~run_with
        ~input_normal_bam:normal ~input_tumor_bam:tumor
        ~result_prefix `Map_reduce in
    somatic_variant_caller
      {Variant_caller.name = "Mutect";
       configuration_json;
       configuration_name;
       make_target;}
      bam_pair
```
I think that this is bad because
1. it is biolerplate
2. why is it in `pipeline.ml` as opposed to `mutect.ml`? 

I think this limitation is because the `Pipeline.t` isn't further parameterized by the configuration of the underlying tools that it needs. Doing this would allow you to just use the `somatic_variant_caller` constructor, and the `Compiler` step will pattern match to the appropriate variant caller based on configuration.

I think you can make an argument that you want nicer, easier to read, construction functions (ie. `strelka`, `mutect` ..etc), but I would put those into a separate module (ie. `Pipeline.Specific` ?). Once another hla typer is added, I would make this distinction.